projects/packages/classic-theme-helper/src/custom-post-types/class-jetpack-portfolio.php

```diff
@@ -59,7 +59,14 @@ public function __construct() {
 			// If called via REST API, we need to register later in lifecycle.
 			add_action( 'restapi_theme_init', array( $this, 'maybe_register_cpt' ) );
 
-			$this->maybe_register_cpt();
+			// If portfolio cpt is enabled, hook into init to register the CPT, otherwise run maybe_register_cpt immediately to deregister.
+			if ( get_option( self::OPTION_NAME, '0' ) ) {
+				$this->maybe_register_cpt();
+
+			} else {
+				add_action( 'init', array( $this, 'maybe_register_cpt' ) );
+
+			}
```

Comment:
To explain the extra logic in `__construct`: We immediately run `maybe_register_cpt` if the portfolio option exists as it needs to be registered immediately. If it doesn't exist, we hook the function call into init to ensure it's only cleaned up after WP has fully loaded. 

In short this is to ensure `maybe_register_cpt` is run at the correct lifecycle stage when the portfolio option exists vs when it doesn't, as this was a noticeable issue when testing the feature via the package.